### PR TITLE
loans added

### PR DESCRIPTION
**Pull request by clementbalestrat**

### CODE REVIEW

File: src/loans-mapping.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  LoanCreated as LoanCreatedEvent,
+  LoanClosed as LoanClosedEvent,
+} from '../generated/EtherCollateral/EtherCollateral';
+
+import { Loan, LoanCreated, LoanClosed } from '../generated/schema';
+
+export function handleLoanCreated(event: LoanCreatedEvent): void {
+  let loanId = event.params.loanID;
+  let loanEntity = new Loan(loanId.toHex());
+  let loanCreatedEntity = new LoanCreated(loanId.toHex());
```
Comment (justinjmoses) [inline src/loans-mapping.ts:11]: this ID should be the transaction hash and log index cause it's the event not the entity

File: src/loans-mapping.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+  loanCreatedEntity.account = event.params.account;
+  loanCreatedEntity.amount = event.params.amount;
+  loanCreatedEntity.save();
+}
+
+export function handleLoanClosed(event: LoanClosedEvent): void {
+  let loanId = event.params.loanID;
+  let loanEntity = Loan.load(loanId.toHex());
+  let loanClosedEntity = new LoanClosed(loanId.toHex());
```
Comment (justinjmoses) [inline src/loans-mapping.ts:26]: (as above)

File: src/loans-mapping.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  loanCreatedEntity.loanId = event.params.loanID;
+  loanCreatedEntity.timestamp = event.block.timestamp;
+  loanCreatedEntity.save();
+}
+
+export function handleLoanClosed(event: LoanClosedEvent): void {
+  let loanEntity = Loan.load(event.params.loanID.toHex());
+  let loanClosedEntity = new LoanClosed(event.transaction.hash.toHex() + '-' + event.logIndex.toString());
+
+  loanEntity.isOpen = false;
```
Comment (justinjmoses) [inline src/loans-mapping.ts:29]: maybe add a `closedAt` field as well?

Comment (clementbalestrat) [reply src/loans-mapping.ts:29]: yeah actually

**Pull request merged**